### PR TITLE
fix : implement settlement api security with privileges

### PR DESCRIPTION
**Pull request by hsuyeemon-tw**

### CODE REVIEW

File: packages/domain-lib/src/privileges.ts [hunk 1]
```diff
@@ -33,9 +33,9 @@ export enum Privileges {
 	// CREATE_SETTLEMENT_BATCH = "SETTLEMENTS_CREATE_BATCH",
 	// CREATE_SETTLEMENT_TRANSFER = "SETTLEMENTS_CREATE_TRANSFER",
 
-	CREATE_DYNAMIC_SETTLEMENT_MATRIX = "SETTLEMENTS_CREATE_DYNAMIC_MATRIX",
-	CREATE_STATIC_SETTLEMENT_MATRIX = "SETTLEMENTS_CREATE_STATIC_MATRIX",
```
Comment (bushjames) [inline packages/domain-lib/src/privileges.ts:37]: is there no longer a need for static settlement matrices?

**Review by bushjames — CHANGES_REQUESTED** [review]
Just a few code comments to remove and a question.

File: packages/domain-lib/src/privileges.ts (hunk 1 above)
Comment (bushjames) [reply packages/domain-lib/src/privileges.ts:37]: Just to clarify, I saw the static matrix privilege removed and now just a general privilege to create a matrix. Does this mean there is now one privilege for both dynamic and static matrices?

File: packages/api-svc/src/routes.ts [hunk 2]
```diff
@@ -149,86 +153,102 @@ export class ExpressRoutes {
...
+		return false;
+	}
+
 	get MainRouter(): express.Router {
 		return this._router;
 	}
 
-	private async getSettlementModels(req: express.Request, res: express.Response): Promise<void>{
+	private async getSettlementModels(req: express.Request, res: express.Response): Promise<void> {
 		// TODO enforce privileges
```
[the remark this answers is not in the document]
Comment (hsuyeemon-tw) [reply packages/api-svc/src/routes.ts:207]: Sure , James. I will remove those comments.

File: packages/domain-lib/src/privileges.ts (hunk 1 above)
Comment (hsuyeemon-tw) [reply packages/domain-lib/src/privileges.ts:37]: Yes, we will use the same Settlement create privilege for both static and dynamic matrix for creation.

**Comment by hsuyeemon-tw** [comment]
@koekiebox Can you please review this PR to in line with your previous changes in settlement-bc ?

**Comment by koekiebox** [comment]
> @koekiebox Can you please review this PR to in line with your previous changes in settlement-bc ?

@hsuyeemon-tw , please see the quality gate issues.

**Review by koekiebox — CHANGES_REQUESTED** [review]
LGTM, please see the Sonarcloud reportings.

**Comment by hsuyeemon-tw** [comment]
hi @koekiebox , I have updated the codes according to the sonar cloud reporting. Can we merge this PR?

**Review by koekiebox — APPROVED** [review]
LGTM.

File: packages/api-svc/src/service.ts [hunk 3]
```diff
@@ -184,17 +189,18 @@ export class Service {
 			);
 			await (logger as KafkaLogger).init();
 		}
-		globalLogger = this.logger = logger;
 
+		
+		globalLogger = this.logger = logger;
```
Comment (pedrosousabarreto) [inline packages/api-svc/src/service.ts:194]: This goes with the top block (logger), should stay where it was.

File: packages/api-svc/src/service.ts [hunk 4]
```diff
@@ -215,9 +221,10 @@ export class Service {
 				authRequester,
 				messageConsumer
 			);
+
 			// MUST only add privileges once, the cmd handler is already doing it
-			// addPrivileges(authorizationClient as AuthorizationClient);
-			// await (authorizationClient as AuthorizationClient).bootstrap(true);
+			authorizationClient.addPrivilegesArray(SettlementPrivilegesDefinition);
```
Comment (pedrosousabarreto) [inline packages/api-svc/src/service.ts:226]: `addPrivilegesArray()` and `bootstrap()`  cannot be done in the api-svc, only the command-handler.-svc - please the comment in the line 225 above.

**Review by pedrosousabarreto — CHANGES_REQUESTED** [review]
Looks good, left a couple of change requests.
Focused only on code structure, @koekiebox pls check for functional.

**Review by pedrosousabarreto — APPROVED** [review]
LGTM

**Pull request merged**